pubpub/platform
# Add page-level access checks and remove integrations page

## Issue(s) Resolved

Resolves #637
Resolves #801

## High-level Explanation of PR


This PR extends the `userCan` function introduced in #671 to return whether or not a user has the specified **community** capability.

In addition, this PR restricts access to certain pages based on a user's `editCommunity` capability (which is only conferred by a community `edit` or `admin` role).

For the purposes of this PR, "restricts access" means that links to those restricted pages should be hidden if the user does not have `edit` or `admin` role for the visited community. And if a user directly navigates to a restricted page, they will see an "Unauthorized" page.

I've also rolled in the issue to remove the integrations links/pages because the change was so small.

## Test Plan

- Add a member to a community with the "contributor" role.
- Log in as the user.
- Observe that nav links to the following pages are hidden:
  * Stage Editor
  * Types
  * Fields
  * Forms
  * Integrations
  * Members
  * Settings
 - Manually navigate to each of the above pages. You should see an "Unauthorized" error page.
 - Grant the user the `editor` role. Observe the above nav links are visible and that the pages render as normal.
 - Grant the user the `admin` role. Observe the above nav links are visible and that the pages render as normal.

## Screenshots (if applicable)

## Notes
